tools/testing/test_selections.py

```diff
@@ -116,8 +116,12 @@ def calculate_shards(
 
 def _query_changed_test_files() -> List[str]:
     default_branch = f"origin/{os.environ.get('GIT_DEFAULT_BRANCH', 'main')}"
-    cmd = ["git", "diff", "--name-only", default_branch, "HEAD"]
-    proc = subprocess.run(cmd, capture_output=True)
+    merge_base = subprocess.run(
+        ["git", "merge-base", default_branch, "HEAD"], capture_output=True
+    )
```

Comment:
`subprocess.run` returns `CompletedProcess` instance, but you need an output string, don't you?
```suggestion
    merge_base = subprocess.check_output(
        ["git", "merge-base", default_branch, "HEAD"]
    ).decode().strip()
```